src/math_utils.rs

```diff
@@ -108,11 +136,11 @@ pub fn ec_double(point: (BigInt, BigInt), alpha: &BigInt, prime: &BigInt) -> (Bi
 /// the given point.
 /// Assumes the point is given in affine form (x, y) and has y != 0.
 pub fn ec_double_slope(point: (BigInt, BigInt), alpha: &BigInt, prime: &BigInt) -> BigInt {
-    assert!(point.1.clone() % prime != bigint!(0));
+    assert!(point.1.clone() % prime != bigint!(0_i32));
     div_mod(
-        bigint!(3) * point.0.clone() * point.0.clone() + alpha,
-        bigint!(2) * point.1,
-        prime.clone(),
+        &(bigint!(3_i32) * point.0.clone() * point.0.clone() + alpha),
+        &(bigint!(2_i32) * point.1),
```

Comment:
Avoid clones and use `is_zero` for checks.

Reply:
resolved:
- https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/a0008d67fc3cf60bc34e3cd8d19773e833d52772
- https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/83bde4c8487d60e10b3bd34bc4e9b7cb23857f5c